services/utilServices/AuthService.js

```diff
@@ -22,10 +25,12 @@ class AuthService {
   }
 
   async getGithubAuthToken({ csrfState, code, state }) {
-    const isCsrfValid =
-      jwtUtils.verifyToken(csrfState) &&
-      jwtUtils.decodeToken(csrfState).state === state
-    if (!isCsrfValid) throw new ForbiddenError()
+    try {
+      const decoded = jwtUtils.verifyToken(csrfState)
+      if (decoded.state !== state) throw new Error("State does not match")
```

Comment:
Good catch, but I was wondering if we should be providing this information? Not sure what the original intention was in relation to the identity PR, but I think it might be alright if users get a generic `ForbiddenError` if this step fails, what are your thoughts?

Reply:
hmm upon closer inspection i think we might have missed out the decoding step.

Reply:
i think the way this is written allows us to catch errors from `verifyToken/decodeToken` and transform it into a generic `ForbiddenError` - i'm assuming this is because we want the `errorHandler` middleware to return a useful message (to the client) instead of the internal error messages that the library behind `jwtUtils` throw.

if this is the case, i'm fine w/ this as-is but preferably document this down as a comment because the code structure right now is not as easy to follow

Reply:
Decoding is actually done by jwt.verify already ([link](https://www.npmjs.com/package/jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback))!

Agreed with the difficulty in following, I've added a comment for it. Thanks for investigating!

Reply:
Sure, 643a00c